.github/workflows/linux.yml

```diff
@@ -9,12 +9,22 @@ jobs:
       matrix:
         cxx: [g++-4.8, g++-8, g++-10, clang++-9]
         build_type: [Debug, Release]
+        standard: [11, 14, 17, 20]
```

Comment:
We should limit the number of configs to make this manageable. 14 is definitely not needed and the rest should be trimmed considerably.

Reply:
Can you help me understand what aspects of this you see as unmanageable so that I can arrive at an agreeable solution?

Depending on how thin you'd like to cut down the tested configurations, I'm considering something more whitelist-y than this approach. Roughly:

```
matrix:
  build_type: [Debug, Release]
  include:
    - cxx: gcc-4.8
      standard: 11
    - cxx: gcc-8
      standard: 14
    - cxx: gcc-10
      standard: 17
    - cxx: gcc-10
      standard: 20
    - cxx: clang-9
      standard: 20
```

Reply:
Per https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-running-with-more-than-one-operating-system that approach would allow this PR to integrate @alexezeder and @darklukee's suggestion of using varied OSs.

Reply:
> Can you help me understand what aspects of this you see as unmanageable so that I can arrive at an agreeable solution?

Too many configs. We should only add ones that are likely to provide high CI signal, ideally one config per std, not compiler-version * std.

>  I'm considering something more whitelist-y than this approach.

LGTM

Reply:
As explained here https://github.com/fmtlib/fmt/pull/1999#issuecomment-727259874, the approach I suggested does not work as expected. The current approach seems to be the only way to achieve the end goal of testing multiple compilers at 1 or more standards in both `Debug` and `Release` mode.